examples/inspection/plot_permutation_importance.py

```diff
@@ -121,59 +130,106 @@
 #   therefore do not reflect the ability of feature to be useful to make
 #   predictions that generalize to the test set (when the model has enough
 #   capacity).
-ohe = rf.named_steps["preprocess"].named_transformers_["cat"]
-feature_names = ohe.get_feature_names_out(categorical_columns)
-feature_names = np.r_[feature_names, numerical_columns]
+#
+# The latter point explains why both the `random_num` and `random_cat` features
+# have a non-null importance. The former point explains why the `random_num`
+# has a really large importance in comparison with `random_cat` while we would
+# expect both random features to have a null importance.
+import pandas as pd
 
-tree_feature_importances = rf.named_steps["classifier"].feature_importances_
-sorted_idx = tree_feature_importances.argsort()
+feature_names = rf[:-1].get_feature_names_out()
 
-y_ticks = np.arange(0, len(feature_names))
-fig, ax = plt.subplots()
-ax.barh(y_ticks, tree_feature_importances[sorted_idx])
-ax.set_yticks(y_ticks)
-ax.set_yticklabels(feature_names[sorted_idx])
-ax.set_title("Random Forest Feature Importances (MDI)")
-fig.tight_layout()
-plt.show()
+mdi_importance = pd.Series(
+    rf[-1].feature_importances_, index=feature_names
+).sort_values(ascending=True)
+
+# %%
+mdi_importance.plot.barh()
+plt.title("Random Forest Feature Importances (MDI)")
+_ = plt.xlabel("Mean decrease in impurity")
 
 
 # %%
 # As an alternative, the permutation importances of ``rf`` are computed on a
 # held out test set. This shows that the low cardinality categorical feature,
-# ``sex`` is the most important feature.
+# `sex` and `pclass` are the most important feature. Indeed, permuting the
+# values of these features will lead to most decrease in accuracy score of the
+# model on the test set.
 #
 # Also note that both random features have very low importances (close to 0) as
 # expected.
+from sklearn.inspection import permutation_importance
+
 result = permutation_importance(
-    rf, X_test, y_test, n_repeats=10, random_state=42, n_jobs=2
+    rf[-1], rf[:-1].transform(X_test), y_test, n_repeats=10, random_state=42, n_jobs=2
 )
-sorted_idx = result.importances_mean.argsort()
 
-fig, ax = plt.subplots()
-ax.boxplot(
-    result.importances[sorted_idx].T, vert=False, labels=X_test.columns[sorted_idx]
+sorted_importances_idx = result.importances_mean.argsort()
+importances = pd.DataFrame(
+    result.importances[sorted_importances_idx].T,
+    columns=feature_names[sorted_importances_idx],
 )
-ax.set_title("Permutation Importances (test set)")
-fig.tight_layout()
-plt.show()
+importances.plot.box(vert=False, whis=10)
+plt.title("Permutation Importances (test set)")
+plt.axvline(x=0, color="k", linestyle="--")
+_ = plt.xlabel("Decrease in accuracy score")
```

Comment:
The `y_ticklabels` are cut off: (See `ndom_cat` and `dom_num`)

![Screen Shot 2022-04-08 at 5 34 38 PM](https://user-images.githubusercontent.com/5402633/162535253-4098d49f-60d2-4c4f-a988-da22fbc212af.jpg)